pennylane/capture/dynamic_shapes.py

```diff
@@ -50,16 +50,26 @@ def _get_shape_for_array(x, abstract_shapes: list) -> dict:
     ``abstract_shapes`` contains all the tracers found in shapes.
 
     """
+    if getattr(x, "shape", None) == () and "int" in str(getattr(x, "dtype", None)):
```

Comment:
```suggestion
    if getattr(x, "shape", None) == () and qml.math.issubdtype(x, "int"):
```

Optional suggestion. I tested it locally and it works (after importing `qml`), but it might fail for other non-tested cases

Reply:
```suggestion
    if getattr(x, "shape", None) == () and jax.numpy.issubdtype(getattr(x, "dtype", None), "int")
```
Thanks for the suggestion. I can just use the jax version here.